kazoo/handlers/utils.py

```diff
@@ -241,7 +243,7 @@ def create_tcp_connection(
             # verify_mode to CERT_NONE.
             # TODO: Make hostname verification configurable as some users may
             # elect to use it.
-            context.check_hostname = False
+            context.check_hostname = check_hostname
             context.verify_mode = (
                 ssl.CERT_REQUIRED if verify_certs else ssl.CERT_NONE
```

Comment:
Based on the comment L242, it seems we might need to raise a `ValueError` or something if `check_hostname=True` and `verify_certs=False`. Maybe in the client class when checking the given params. What do you think?

Reply:
Agreed. I've added an error for this case.